pkg/validation-webhook/snapshot_test.go

```diff
@@ -0,0 +1,349 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"testing"
+
+	volumesnapshotv1beta1 "github.com/kubernetes-csi/external-snapshotter/client/v2/apis/volumesnapshot/v1beta1"
+	v1 "k8s.io/api/admission/v1"
+	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestAdmitVolumeSnapshot(t *testing.T) {
+	pvcname := "pvcname1"
+	mutatedField := "changed-immutable-field"
+	contentname := "snapcontent1"
+	volumeSnapshotClassName := "volume-snapshot-class-1"
+	testCases := []struct {
```

Comment:
put the verb as the first word of each test case to make it clear?
i.e.,
"Delete - both new and old are nil"
"Update - old invalid and new valid"